sdk/schemaregistry/azure-schemaregistry-avroencoder/azure/schemaregistry/encoder/avroencoder/_schema_registry_avro_encoder.py

```diff
@@ -0,0 +1,276 @@
+        :paramtype message_type: MessageCallbackType or MessageType or None
+        :rtype: MessageType or MessageMetadataDict
+        :raises ~azure.schemaregistry.encoder.avroencoder.exceptions.SchemaParseError:
+            Indicates an issue with parsing schema.
+        :raises ~azure.schemaregistry.encoder.avroencoder.exceptions.SchemaEncodeError:
+            Indicates an issue with encoding data for provided schema.
+        """
+
+        raw_input_schema = schema
+
+        try:
+            schema_fullname = self._avro_encoder.get_schema_fullname(raw_input_schema)
+        except Exception as e:  # pylint:disable=broad-except
+            SchemaParseError(
+                f"Cannot parse schema: {raw_input_schema}", error=e
+            ).raise_with_traceback()
+
+        http_request_kwargs = get_http_request_kwargs(kwargs)
+        schema_id = self._get_schema_id(schema_fullname, raw_input_schema, **http_request_kwargs)
+        content_type = f"{AVRO_MIME_TYPE}+{schema_id}"
+
+        try:
+            data_bytes = self._avro_encoder.encode(value, raw_input_schema)
+        except Exception as e:  # pylint:disable=broad-except
+            SchemaEncodeError(
+                "Cannot encode value '{}' for schema: {}".format(
+                    value, raw_input_schema
+                ),
+                error=e,
+            ).raise_with_traceback()
+
+        stream = BytesIO()
+
+        stream.write(data_bytes)
+        stream.flush()
+
+        payload = stream.getvalue()
+        stream.close()
+
+        if message_type:
+            try:
+                return message_type(data=payload, content_type=content_type, **kwargs)
+            except TypeError as e:
+                SchemaEncodeError(
+                    f"""The data model {str(message_type)} does not support MessageCallbackType protocol.
+                        If using an Azure SDK model class, please check the README.md for the full list
+                        of supported Azure SDK models and their corresponding versions."""
+                ).raise_with_traceback()
+
+        return {"data": payload, "content_type": content_type}
+
+    def _convert_preamble_format(
+        self, data, content_type
+    ):  # pylint: disable=no-self-use
+        record_format_identifier = b"\0\0\0\0"
+        if data[0:RECORD_FORMAT_IDENTIFIER_LENGTH] == record_format_identifier:
+            schema_id = data[
+                SCHEMA_ID_START_INDEX : (SCHEMA_ID_START_INDEX + SCHEMA_ID_LENGTH)
+            ].decode("utf-8")
+            content_type = f"{AVRO_MIME_TYPE}+{schema_id}"
+            data = data[DATA_START_INDEX:]
+
+        return data, content_type
+
+    def decode(
+        self,
+        *,
+        message: Optional[MessageType] = None,
+        data: Optional[bytes] = None,
+        content_type: Optional[str] = None,
+        **kwargs,
+    ) -> Dict[str, Any]:
```

Comment:
it's just returning `Dict` or `MessageType` as well?

Reply:
`decode` just returns Dict, which is the decoded data. We don't return any of the message types here.